### ISSUE TITLE
Mission JSON

### ISSUE THREAD

**Issue by atar13** [tritonuas/obcpp#65]
## 🚀 Feature
Load mission information from a file (`mission.json`) upon starting the program.

## Motivation

We want the default behavior of the OBC to be that it loads the previous mission automatically upon start up.

## Implementation

The code for the mission config is located in these two files:

- `include/core/config.hpp`
- `src/core/config.cpp`

Here are the steps I would follow to complete this task:

1. in `config.hpp`, add a new private class function that takes the current state of the MissionConfig class and writes it to a `mission.json` file.
2. Call this private class function in all of the various setter functions, so that whenever the config gets updated it updates the `mission.json` file.
3. Write the version of the constructor that takes in a filename and reads in the data from that filename. If there is no file, then it should do nothing.
4. Delete the version of the constructor that takes no parameter, then it will probably error in `include/core/states.hpp` or `include/core/states.cpp`, complaining that there is no default constructor for `MissionConfig`. You can then call the new filename constructor inside of the constructor for MissionState, using an [initializer list](https://en.cppreference.com/w/cpp/language/constructor). If you're confused by the C++ syntax then a lead or someone can help

## Note on Protobufs

We are using something called protobufs, which are a convenient way to manage passing messages around different programs/servers. If you want the long explanation you can ask someone in person, but here is how this is relevant for this task:

If you have these include statements: 
```cpp
#include "obc.pb.h"
#include <google/protobuf/util/json_util.h>
```

Then you have access to our protobuf specific types, and the function to convert the protobuf class to JSON.

This is how you take a json string (in this case `request.body`) and put it inside of a protobuf class.
```cpp
    Mission mission;
    google::protobuf::util::JsonStringToMessage(request.body, &mission);
```

Basically you will have to do the reverse of this. So you create an object of type `Mission`, set all of the required values with the setter functions they provide, then there should be some equivalent function in the `google::protobuf::util` namespace which takes the Mission and turns it into a json string.

**Comment by Tyler-Lentz** [comment]
RIP Kyle, currently not planned
